pkg/plugin/run.go

```diff
@@ -261,11 +262,23 @@ func (e *AutoscaleEnforcer) handleResources(
 	vCPUTransition := collectResourceTransition(&node.vCPU, &pod.vCPU)
 	memTransition := collectResourceTransition(&node.memSlots, &pod.memSlots)
 
+	if lastPermit != nil {
+		vCPUVerdict := vCPUTransition.handleLastPermit(lastPermit.VCPU)
+		memVerdict := memTransition.handleLastPermit(lastPermit.Mem)
+		logger.Info(
+			"Handled last permit info from pod",
+			zap.Object("verdict", verdictSet{
+				cpu: vCPUVerdict,
+				mem: memVerdict,
+			}),
+		)
+	}
+
 	vCPUVerdict := vCPUTransition.handleRequested(req.VCPU, startingMigration, !supportsFractionalCPU)
 	memVerdict := memTransition.handleRequested(req.Mem, startingMigration, false)
```

Comment:
reusing the `resourceTransition` across multiple function calls smells unsound (iirc, the methods are implemented with the (undocumented) assumption that at the start, the `oldNode` fields are equal to those in `node`, and also for `oldPod` and `pod`).

Was there a particular reason to reuse the `resourceTransition`s here?

---

aside: I think it probably makes sense to split out the `oldNode` and `oldPod` fields into values that are only created & used inside the methods, rather than stored alongside the mutable fields. That would make this sound, and probably easier to understand too.

Reply:
Makes sense. I'm going to rename `resourceTransition` to `resourceTransitioner` because it not an object that represents a transition, but actually it's performs transition on a resource state. I'll remove `oldPod` and `oldNode` from here. I'm just wondering why did you made those inline structs and just didn't make a copy of `nodeResourceState` and `podResourceState`?